source/common/json/json_streamer.h

```diff
@@ -3,15 +3,67 @@
 #include <memory>
 #include <stack>
 #include <string>
+#include <type_traits>
+#include <variant>
 
 #include "envoy/buffer/buffer.h"
 
+#include "source/common/buffer/buffer_util.h"
+#include "source/common/json/json_sanitizer.h"
+
 #include "absl/strings/string_view.h"
 #include "absl/types/variant.h"
 
 namespace Envoy {
 namespace Json {
 
+// To ensure the streamer is being used correctly, we use assertions to enforce
+// that only the topmost map/array in the stack is being written to. To make
+// this easier to do from the Level classes, we provider Streamer::topLevel() as
+// a member function, but this is only needed when compiled for debug.
+//
+// We only compile Streamer::topLevel in debug to avoid having it be a coverage
+// gap. However, assertions fail to compile in release mode if they reference
+// non-existent functions or member variables, so we only compile the assertions
+// in debug mode.
+#ifdef NDEBUG
+#define ASSERT_THIS_IS_TOP_LEVEL                                                                   \
+  do {                                                                                             \
+  } while (0)
+#define ASSERT_LEVELS_EMPTY                                                                        \
+  do {                                                                                             \
+  } while (0)
+#else
+#define ASSERT_THIS_IS_TOP_LEVEL ASSERT(this->streamer_.topLevel() == this)
+#define ASSERT_LEVELS_EMPTY ASSERT(this->levels_.empty())
+#endif
+
+class Constants {
+public:
+  // Constants for common JSON values.
+  static constexpr absl::string_view True = R"(true)";
+  static constexpr absl::string_view False = R"(false)";
+  static constexpr absl::string_view Null = R"(null)";
+
+  // Constants for JSON delimiters.
+  static constexpr absl::string_view MapBeg = R"({)";
```

Comment:
Can you just make this a straight inlining change instead of making a bunch of other changes at the same time?

Reply:
What you mean by `a straight inlining change instead`? Move code only?